tests/test_other.py

```diff
@@ -9820,3 +9820,12 @@ def test_autoconf_mode(self):
     # Test that output name is just `a.out` and that it is directly executable
     output = self.run_process([os.path.abspath('a.out')], stdout=PIPE).stdout
     self.assertContained('hello, world!', output)
+
+  def test_standalone_export_main(self):
+    # Tests that explictly exported `_main` does not fail.   Since we interpret an
+    # export of `_main` to be be an export of `__start` in standalone mode the
```

Comment:
by whom is main interpreted as start? in lld, or in emscripten? (if in emscripten, where?)

Reply:
In emscripten.   We currently treat _main a little magically in standalone mode.   If we see `_main` we interpret it as "please export your entry point" and not "please export the main function" so that standalone mode and non-standalone mode users can appear more similar.

Perhaps we should change this?   I agree its somewhat confusing.    If we do we could probably include a warning to standalone users if they try to export `_main`.   We could also need to change the default EXPORT_FUNCTIONS list when in standalone more.  It would clean things up though.

Would you rather do this cleanup pro-actively or land this change and cleanup after?

Reply:
There is also more magic in there because we interpret the lack of _main in EXPORTED_FUNCTIONS to mean "i'm want to build a reactor".  This is a convenient synonym for `--no-entry`, but its also a little too magical and maybe we should get rid of that too?

Reply:
I'm probably ok with this change but I don't understand it enough yet. Where exactly in emscripten do we interpret main as start? Searching in emscripten.py, emcc.py, and tools/*.py I don't think I see such a place, so I'm missing something.

Reply:
I think the main (only?) place is in `building.py` : 

Here we remove main from the list of things we ask lld to export: https://github.com/emscripten-core/emscripten/blob/master/tools/building.py#L513

And here we implicitly add `_start` to the list by not building `--no-entry` (_start is the default entry): https://github.com/emscripten-core/emscripten/blob/ca4457fbcebd968587b53415253be8423d33b4c8/tools/building.py#L532

Reply:
Then in postable.js we make assumptions about the name of the entry point accordingly.

Reply:
I see, thanks!